src/twisted/mail/_except.py

```diff
@@ -123,17 +131,14 @@ def __init__(
     def __str__(self) -> str:
         return self.__bytes__().decode("utf-8")
 
-    def __bytes__(self):
+    def __bytes__(self) -> bytes:
         if self.code > 0:
-            res = ["{:03d} {}".format(self.code, self.resp)]
+            res = ["{:03d} {!r}".format(self.code, self.resp).encode("utf-8")]
```

Comment:
Not sure if `!r` here is intended, but it's the current behavior on Py3.
We encode to bytes here so that `res` is `List[bytes]` and also eliminates the need for the loop later.